viewing_party/party.py

```diff
@@ -1,23 +1,360 @@
+    - return the average rating
+    '''
+    #creating basic variables that we'll need
+    average_rating = 0.0
+    total = 0.0
+
+    #if the "watched is empty, we return 0.0"
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    else:
+        #using the for loop to iterate through every movie in "watched"
+        for movie in user_data["watched"]:
+            #and calculate total sum of movie ratings
+            total += movie["rating"]
+        #number of movies equals the length of "watched"    
+        movie_count = len(user_data["watched"])
+        average_rating = total/movie_count
+        return average_rating    
+
+def get_most_watched_genre(user_data):
+    '''
+    - take one parameter: `user_data`
+    - Determine which genre is most frequently occurring in the watched list
+    - return the genre that is the most frequently watched
+    - If the value of "watched" is an empty list, `get_most_watched_genre` should return `None`.
+    
+    '''
+
+    # creating an empty dictionary fo genres to calculate frequency
+    # genre will be a key and frequency a value
+    genres = {} 
+    if len(user_data["watched"]) == 0: #if "watched" is empty, we return None
+        return None
+    else:
+        #using a for loop to iterate through "watched"
+        for movie in user_data["watched"]:
+            #if it's a first time we encounter this genre we create key/value pair with value "1"
+            if movie["genre"] not in genres:
+                genres[movie["genre"]] = 1
+            #if we already have this key, we add 1 to the value
+            else:
+                genres[movie["genre"]] += 1
+        
+        #Here I wanted to use max function and found a method to get the specific key corresponding to this max value.
+        #I'm not checking an edge case where there's two exact same frequencies with maximum value
+        most_watched_genre = max(genres, key=genres.get)
+
+        #we can also do something like this:
+        # frequency_list = genres.values()
+        # max_value = max(frequency_list)
+        # for genre in genres:
+        #     if genres[genre] == max_value:
+        #         most_watched_genre = genre
+                #and this loop might be updated to catch several genres with the same frequency
 
+        return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_user_movies_list(user_data, category):
+    '''
+    helper function get_user_movies_list
+    - creates a list of titles of a requiered category ("watched", "watchlist" or "favorites")
+    - returns list of titles
+    '''
+    #creating empty list
+    movies_list = []
+    #using for loop to iterate through this entire category and add all the titles to the list
+    for movie in user_data[category]:
+        movies_list.append(movie["title"])
+
+    return movies_list

```

Comment:
This is a great helper function. I wouldn't normally advocate for creating just a list of the titles, when we can make a list of the entire movies themselves, but given the way you have parameterized it, I think I can see where you were headed!